### PR TITLE
MCIP 57 update ring uniqueness requirements for SCIs

### PR DESCRIPTION
**Pull request by garbageslam**
See [MCIP 57](https://github.com/mobilecoinfoundation/mcips/pull/57) for discussion

### CODE REVIEW

File: transaction/core/src/validation/validate.rs [hunk 1]
```diff
@@ -181,18 +181,26 @@ pub fn validate_ring_sizes(
...
-        .flat_map(|tx_in| tx_in.ring.iter())
-        .collect();
-
-    check_unique(
-        &ring_elements,
-        TransactionValidationError::DuplicateRingElements,
-    )
+    let mut ring_elements_without_input_rules = HashSet::<&TxOut>::default();
+    for input in tx_prefix.inputs.iter() {
+        if input.input_rules.is_some() {
```
Comment (remoun) [inline transaction/core/src/validation/validate.rs:190]: Consider adding a test case exercising this branch

**Review by remoun — APPROVED** [review]

File: transaction/core/src/validation/validate.rs (hunk 1 above)
Comment (garbageslam) [reply transaction/core/src/validation/validate.rs:190]: 👍

**Review by eranrund — APPROVED** [review]
LGTM once a test case is added like @remoun suggested

**Review by briancorbin — APPROVED** [review]
This LGTM, I'm in favor of making this change based off of the MCIP proposal.

**Pull request merged**